lib/galaxy/schema/schema.py

```diff
@@ -3043,6 +3057,14 @@ class Config:
         extra = Extra.allow  # Allow any other extra fields
 
 
+class CreateNotificationPayload(NotificationBase):
+    notification: Optional[str] = NotificationContentField
```

Comment:
I think you're overcomplicating things here. All you need in the CreateNotificationPayload is the message_text property, i don't think this requires separately defined fields, just do define it here.